### ISSUE TITLE
Support for diagnostics seems unfinished?

### ISSUE THREAD

**Issue by tintoy** [KirillOsenkov/XmlParser#10]
Hi.

Thanks for building a great library, but looking through the code the support for attaching diagnostics to syntax nodes seems incomplete. For example:

https://github.com/KirillOsenkov/XmlParser/blob/be216f0851f6445bb50307997654c7e07595fde9/src/Microsoft.Language.Xml/Scanner.cs#L354-L357

And:

https://github.com/KirillOsenkov/XmlParser/blob/554a2ece6db17ff9e0712648580d1d177617fc51/src/Microsoft.Language.Xml/ErrorFactory.cs#L7-L15

I was hoping to be able to use the diagnostics to quickly detect what (if anything) is broken so I can either repair or omit those nodes.

Are you still planning to add this functionality?

**Comment by KirillOsenkov** [comment]
Yes, it is unfinished unfortunately and in an ideal world I would certainly want to finish it up. Unfortunately, I really don't have time right now :| Maybe later...

If a PR came along with a couple unit-tests, I'd certainly consider taking it :)

**Comment by tintoy** [comment]
I just might do that, then :)
